analysisCode/src/BreitFrame.cpp

```diff
@@ -0,0 +1,17 @@
+#include "include/BreitFrame.h"
+
+void labToBreit( TLorentzVector *l, erhic::EventPythia* tr_ev )
+{
+
+  TLorentzVector q_lab = tr_ev->ExchangeBoson()->PxPyPzE();
+  TLorentzVector P_lab = tr_ev->BeamHadron()->PxPyPzE();
+  double E_breit =  2*tr_ev->GetTrueX()*P_lab.E() + q_lab.E();
+  TVector3 beta(  2*tr_ev->GetTrueX()*P_lab.Vect() + q_lab.Vect() );
+  beta *= 1/E_breit;
+  TVector3 z_axis(0,0,1);
+  l->Boost(-beta);
+  q_lab.Boost(-beta);
+  // Double check signage or if inverse rotation is needed... 
+  l->Rotate(q_lab.Vect().Angle(z_axis), q_lab.Vect().Cross(z_axis).Unit());
```

Comment:
Can you explain some more about the need for this? Why is `*l` boosted and then rotated also?

Reply:
My understanding is that the transformation from the lab frame to the Breit frame in a DIS event is specified by a generalized Lorentz transformation (not just a boost) that not only takes the event to a frame of reference where the exchange boson does not transfer any energy (only momentum), but also such that the exchange boson is aligned along the z-axis for simplicity -- this could be parametrized by a single matrix, but the TLorentzVector class most easily accomplishes this in two steps. First, by boosting to a frame where the exchange boson does not transfer energy (beta = - (2xP + q)), and then by rotating to a frame of reference that preserves this quality and such that the momentum transfer is only in the z-direction. After just the boost, q will not in general be aligned along the z-axis and will depend on angles, which is why the rotation is needed. 

There is some literature on this, including the document I sent to you (I tried to attach it here but it wont let me, probably because I cobbled it together from a bunch of screenshots with pdftk) -- but keep in mind that what is shown in the derivation is  "the complete transformation that takes the z-axis in the HERA frame to the z-axis in the Breit frame". We just want to take any arbitrary vector in the lab frame to the Breit frame (in other words, take the exchange boson in the lab frame to the z-axis in the Breit frame to define the transformation), so the first rotation about the y-axis is not necessary in our case (only the boost and subsequent rotation about the y' axis, which is arbitrary in our case). I can try to explain this more over BlueJeans to clear things up further if it would be helpful.

Reply:
Okay I will take a look at the resource you gave me tonight. But I think your explanation makes sense, basically that just because you boost something doesn't mean that it is aligned on a particular axis. So one must boost to the proper frame, and then rotate to e.g. align the 4 vector with a particular axis that is useful.